.github/workflows/build-op-image.yaml

```diff
@@ -18,41 +19,129 @@ on:
-        with:
-          path: ~/go/pkg/mod
-          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+      - name: Set outputs
+        id: set-outputs
+        run: |
+          echo "IMAGE_NAME=${{ env.GITHUB_IMAGE }}" >> $GITHUB_OUTPUT
+          echo "DOCKER_IMG_NAME=${{env.DOCKER_REPO}}/${{ env.DOCKER_IMAGE }}" >> $GITHUB_OUTPUT
 
+  operator-build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    name: Build Image for Fluent Operator
+    needs: 
+      - build-image-metadata
+    steps:
       - name: Checkout code
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Login to Docker Hub
-        uses: docker/login-action@v3
-        with:
-          username: ${{ secrets.REGISTRY_USER }}
-          password: ${{ secrets.REGISTRY_PASSWORD }}
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
 
       - name: Set up Docker Buildx
-        id: buildx
         uses: docker/setup-buildx-action@v3
 
-      - name: Build and Push Image for Fluent Operator
-        run: |
-          tag=$(cat VERSION | tr -d " \t\n\r")
-          make build-op -e FO_IMG=${{ env.REGISTRY_REPO }}/fluent-operator:$tag
\ No newline at end of file
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          file: ./cmd/fluent-manager/Dockerfile
+          push: true
+          platforms: linux/amd64,linux/arm64
+          tags: ${{ needs.build-image-metadata.outputs.tags }}
+          labels: ${{ needs.build-image-metadata.outputs.labels }}
+
+  scan-operator-image:
+    name: Scan Docker Image
+    needs: 
+      - operator-build
+      - build-image-metadata
+    uses: ./.github/workflows/scan-docker-image-action.yaml
+    with:
+      source_image: "${{ needs.build-image-metadata.outputs.IMG_NAME }}:${{ needs.build-image-metadata.outputs.version }}"
+      source_registry: ghcr.io
+      platforms: "['linux/arm64', 'linux/amd64']"
+    secrets:
+      registry_username: ${{ github.actor }}
+      registry_password: ${{ secrets.GITHUB_TOKEN }}
+
+  release-image-to-docker-hub:
+    if: ${{ github.event_name != 'pull_request' }}
+    name: Release Image to Docker Hub
+    uses: ./.github/workflows/clone-docker-image-action.yaml
+    needs: 
+      - operator-build
+      - scan-operator-image
+      - build-image-metadata
+    with:
+      source_image: "${{ needs.build-image-metadata.outputs.IMG_NAME }}:${{ needs.build-image-metadata.outputs.version }}"
+      source_registry: ghcr.io
+      target_image: "${{ needs.build-image-metadata.outputs.DOCKER_IMG_NAME }}:${{ needs.build-image-metadata.outputs.version }}"
+      target_registry: docker.io
+      platforms: "['linux/arm64', 'linux/amd64']"
+    secrets:
```

Comment:
Can we sync the latest image to dockerhub as well?

only the image with branch tag like `kubesphere/fluent-operator:master` is pushed to dockerhub:
![image](https://github.com/fluent/fluent-operator/assets/18525465/381b4490-eada-48cd-88e3-b21e0c7c8b36)
https://github.com/fluent/fluent-operator/actions/runs/8199464406/job/22425032049#step:2:100

The latest image is actually created as well:
https://github.com/fluent/fluent-operator/actions/runs/8199464406/job/22424977637#step:6:382
![image](https://github.com/fluent/fluent-operator/assets/18525465/46c8c234-0b23-4d7f-9c26-dc732e471cbb)

Reply:
> it's ok to leave the latest on ghcr only, we can add a note in the repo's readme saying users can fetch latest image from ghcr if they want .
> 
> And of course sync the latest to dockerhub is also reasonable

As per this message I thought to update the readme that latest tag is available in GHCR and only tagged images are in docker hub. If not I can update the action to include latest tag to all the new images.

Reply:
I realized that syncing the latest image to dockerhub is necessary because the helm and yaml of the master branch is using the latest tag @sarathchandra24 

https://github.com/fluent/fluent-operator/blob/master/charts/fluent-operator/values.yaml#L26-L27
https://github.com/fluent/fluent-operator/blob/master/manifests/setup/fluent-operator-deployment.yaml#L42

Reply:
Created the following PR https://github.com/fluent/fluent-operator/pull/1086